include/ginkgo/core/base/array.hpp

```diff
@@ -637,6 +637,46 @@ class array {
      */
     const value_type* get_const_data() const noexcept { return data_.get(); }
 
+    /**
+     * Returns a single value from the array.
+     *
+     * This involves a bounds check, polymorphic calls and potentially a
+     * device-to-host copy, so it is not suitable for accessing many elements
+     * in performance-critical code.
+     *
+     * @param index  the array element index.
+     * @return the value at index.
+     */
+    value_type get_value(size_type index) const
```

Comment:
This will always involve a device to host copy. So, IMO, the function name is misleasing and has potential for misuse. 

I dont see any applications that would need to use this. I am not sure it is necessary to have this as a member of the `gko::array` class. As I see it, the main benefit is to simplify the verbosity of `exec->copy_val_to_host(...)`. Therefore, I would suggest making it a `static` function in a ginkgo internal header and not expose it in the public interface.

Reply:
I replaced them by `load/store_value`, which seemed to be the favored approach